src/libopenrave-core/jsonparser/jsonreader.cpp

```diff
@@ -133,6 +141,424 @@ static std::string CanonicalizeURI(const std::string& suburi, const std::string&
     return suburi;
 }
 
+
+// Need to remove fragement to check for .json or .msgpack
+static std::string _GetPath(const std::string& uri)
+{
+    std::string path = uri;
+    size_t hashindex = path.find_last_of('#');
+
+    if (hashindex != std::string::npos) {
+        path = path.substr(0, hashindex);
+    }
+    return path;
+}
+
+
+class JSONRemoteHelper
```

Comment:
@JosephBerman I think you can even pull this out to a separate file in the `jsonreader`, e.g. `jsonreader/jsondownloader.cpp`